robot/impl/resource_manager.go

```diff
@@ -817,6 +817,14 @@ func (manager *resourceManager) markResourceForUpdate(name resource.Name, conf r
 	return nil
 }
 
+func (manager *resourceManager) validateUniqueResource(resources map[string]bool, resourceName string) error {
+	if _, ok := resources[resourceName]; ok {
+		return errors.Errorf("resource %s already exists in the config", resourceName)
```

Comment:
```suggestion
		return errors.Errorf("resource %s already exists on the robot", resourceName)
```
[super-nit] there are some builtin resources that exist on the robot but do not exist in the user's config. I assume if a user tried to add a resource called "builtin" of the wrong API they would get this error and be confused.